internal/service/ec2/ec2_eip.go

```diff
@@ -72,8 +73,12 @@ func ResourceEIP() *schema.Resource {
 				Optional: true,
 			},
 			"domain": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:          schema.TypeString,
+				ForceNew:      true,
```

Comment:
Since this and `vpc` are both marked `ForceNew`, will it destroy an existing EIP if someone updates their config from `vpc = true` to `domain = "vpc"`?

Reply:
good call. It does not. I went ahead and added a test to verify

```console
$ make testacc TESTS="TestAccEC2EIP_migrateVPCToDomain" PKG=ec2

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2EIP_migrateVPCToDomain'  -timeout 180m
--- PASS: TestAccEC2EIP_migrateVPCToDomain (30.98s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	34.189s
```